Remotion/Validation/Core/Validators/RegularExpressionValidator.cs

```diff
@@ -1,35 +1,53 @@
-﻿// Decompiled with JetBrains decompiler
-// Type: FluentValidation.Validators.RegularExpressionValidator
-// Assembly: FluentValidation, Version=5.0.0.1, Culture=neutral, PublicKeyToken=a82054b837897c66
-// MVID: 30628A95-CE3F-41E4-BA2A-29882CBD79CE
-// Assembly location: C:\Development\Remotion\trunk-svn2git\packages\FluentValidation-Signed.5.0.0.1\lib\Net40\FluentValidation.dll
-
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
 using System;
 using System.Text.RegularExpressions;
+using JetBrains.Annotations;
+using Remotion.Utilities;
 using Remotion.Validation.Implementation;
 
 namespace Remotion.Validation.Validators
 {
   public class RegularExpressionValidator : PropertyValidator, IRegularExpressionValidator
   {
-    private readonly string _expression;
+    public string Expression { get; }
+
     private readonly Regex _regex;
 
-    public RegularExpressionValidator (string expression, IValidationMessage validationMessage = null)
+    public RegularExpressionValidator ([NotNull] string expression, [CanBeNull] IValidationMessage validationMessage = null)
         : base (Constants.RegularExpressionError, validationMessage ?? new InvariantValidationMessage (Constants.RegularExpressionError))
     {
-      _expression = expression;
+      ArgumentUtility.CheckNotNull ("expression", expression);
+
+      Expression = expression;
       _regex = new Regex (expression);
     }
 
     protected override bool IsValid (PropertyValidatorContext context)
     {
-      return context.PropertyValue == null || _regex.IsMatch ((string) context.PropertyValue);
-    }
+      if (context.PropertyValue == null)
+        return true;
 
-    public string Expression
-    {
-      get { return _expression; }
+      if (context.PropertyValue is string stringValue)
```

Comment:
test for string and return true if not string. in cases such as these, always do the abort-conditions first.